internal/storage/v1/elasticsearch/factory.go

```diff
@@ -65,38 +49,40 @@ type Factory struct {
 	client atomic.Pointer[es.Client]
 
 	pwdFileWatcher *fswatcher.FSWatcher
+
+	templateBuilder es.TemplateBuilder
 }
 
-// NewFactory creates a new Factory.
-func NewFactory() *Factory {
-	return &Factory{
+// NewFactoryBase creates a new FactoryBase.
+func NewFactoryBase() *FactoryBase {
+	return &FactoryBase{
 		Options:     NewOptions(primaryNamespace),
 		newClientFn: config.NewClient,
 		tracer:      otel.GetTracerProvider(),
 	}
 }
 
-func NewArchiveFactory() *Factory {
-	return &Factory{
+func NewArchiveFactoryBase() *FactoryBase {
+	return &FactoryBase{
 		Options:     NewOptions(archiveNamespace),
 		newClientFn: config.NewClient,
 		tracer:      otel.GetTracerProvider(),
 	}
 }
 
-func NewFactoryWithConfig(
+func NewFactoryBaseWithConfig(
 	cfg config.Configuration,
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
-) (*Factory, error) {
+) (*FactoryBase, error) {
 	if err := cfg.Validate(); err != nil {
 		return nil, err
 	}
 
 	defaultConfig := DefaultConfig()
 	cfg.ApplyDefaults(&defaultConfig)
```

Comment:
I don't like this here, it's breaking the pattern used in V2. DefaultConfig is called much earlier, when instantiating the config. Once instantiated the factory should just take it, not re-apply the defaults.